### ISSUE TITLE
5.0.2 crashes often when trying to edit previous transactions

### ISSUE THREAD

**Issue by tomaifauchai** [PaulSquires/TradeTracker#49]
I'm trying several times in a row to edit that trade "My goal is to delete this leg.."

![image](https://github.com/user-attachments/assets/36aa5b83-0393-4bfd-a230-380809fdf257)

![image](https://github.com/user-attachments/assets/992acb38-5758-4c7c-9630-8cb764ef647f)

This just crash every times. I run it from C:\tradetracker-demo\
I tried run as admin, not run as admin, connected to IB, not connected to IB with same results all times.

I don't think its new to 5.0.2, some crashes occured in previous releases and i think this is where the app crash the most. (When i try to edit previous legs.)
The other case would be sometimes when i am connected to IB and i restart IB and i try to reconnect the app to it, i can hear the CPU fan increasing and i know it'll crash anyway so i restart the app. ;)

**Comment by tomaifauchai** [comment]
Ok so when i took this path it did work. But never the "normal" pencil way.
![image](https://github.com/user-attachments/assets/a7e07251-9cb7-4d35-a41a-da607798fdef)

**Comment by PaulSquires** [comment]
Thanks for the report. Not sure yet what would cause such a crash. Are you able to post or email me the tt- database.db file? Especially for the scenario when the edit crashes.

**Comment by PaulSquires** [comment]
Oh, and also, when does the crash happen. As soon as you click the pencil icon?

**Comment by tomaifauchai** [comment]
> Oh, and also, when does the crash happen. As soon as you click the pencil icon?

Correct. Click the pencil then the App shuts down. It happens randomly but this time was reproduced about 10 times in a row until i used the little arrow on the side.

**Comment by PaulSquires** [comment]
I have been able to reproduce the crash. Seems to happen if you start the program, switch to the Transactions tab, and then click on the pencil edit icon. Maybe there are also other combinations that trigger the crash but this one seems to be consistent. I am assuming that a pointer is not being initialized correctly when the first item in the Transactions list is shown for the first time. I will investigate and fix tomorrow.

**Comment by tomaifauchai** [comment]
That sequence sounds familiar to me ! I can also see one sequence when trying to connect to IBKR (paper account) in my case after a TWS restart. But also random sometimes. I can live with this one as it happens maybe 2 or 3 times a week only so far and seems less recurrent than it was in version 4.

Thanks for your help

**Comment by PaulSquires** [comment]
Okay, found the error. 

Line #170 of the file "transaction_edit.cpp" I was testing the true/false boolean "state.is_transactions_data_loaded " rather than "state.is_transedit_data_loaded"

I will recompile everything tomorrow, create the package distributions, and provide new version downloads (version 5.03).

**Comment by tomaifauchai** [comment]
Thanks Paul! I can reproduce the TWS connect bug if you want i can open a new issue for it specifically.

**Comment by PaulSquires** [comment]
> Thanks Paul! I can reproduce the TWS connect bug if you want i can open a new issue for it specifically.

Yes, probably a new bug report would be a good idea because it might take a while before I can fix this issue. It's been a low priority issue that I've had in the "_TODO.txt" file for a long time!

> - Disconnect and then reconnect does not reset state within the Client variable. It does restart the
> threads however.

I just need to dedicate some time to analyze the problem and fix it.

**Comment by PaulSquires** [comment]
Version 5.03 with binary packages has been uploaded and should fix this edit bug issue.
